src/uu/seq/src/seq.rs

```diff
@@ -329,17 +206,85 @@ pub fn uu_app() -> App<'static, 'static> {
         )
 }
 
-fn done_printing<T: Num + PartialOrd>(next: &T, increment: &T, last: &T) -> bool {
+fn done_printing<T: Zero + PartialOrd>(next: &T, increment: &T, last: &T) -> bool {
     if increment >= &T::zero() {
         next > last
     } else {
         next < last
     }
 }
 
+/// Write a big decimal formatted according to the given parameters.
+///
+/// This method is an adapter to support displaying negative zero on
+/// Rust versions earlier than 1.53.0. After that version, we should be
+/// able to display negative zero using the default formatting provided
+/// by `-0.0f32`, for example.
+fn write_value_float(
+    writer: &mut impl Write,
+    value: &ExtendedBigDecimal,
+    width: usize,
+    precision: usize,
+    is_first_iteration: bool,
+) -> std::io::Result<()> {
+    let value_as_str = {
+        let s = if *value == ExtendedBigDecimal::MinusZero && is_first_iteration {
+            format!(
+                "-{value:>0width$.precision$}",
+                value = value,
+                width = if width > 0 { width - 1 } else { width },
+                precision = precision,
+            )
+        } else {
+            format!(
+                "{value:>0width$.precision$}",
+                value = value,
+                width = width,
+                precision = precision,
+            )
+        };
+        if *value == ExtendedBigDecimal::MinusZero && !s.starts_with('-') {
```

Comment:
Is there a reason why we need this additional check? I would assume the check above would've already handled the negative zero case.

Reply:
Let me double check and get back to you, it's been too long since I wrote this so I don't remember.

Reply:
I think you are right, it can be removed. It must have been a leftover from an earlier version of the code that I was working with. I'll remove it.